core-bundle/src/Image/ImageFactory.php

```diff
@@ -326,8 +326,8 @@ private function createImportantPart(ImageInterface $image): ?ImportantPart
         }
 
         if (
-            (float) $file->importantPartX + (float) $file->importantPartWidth > 1
-            || (float) $file->importantPartY + (float) $file->importantPartHeight > 1
+            (float) $file->importantPartX + (float) $file->importantPartWidth >= 2
+            || (float) $file->importantPartY + (float) $file->importantPartHeight >= 2
```

Comment:
You could probably also check if all values are integers to avoid the constant. :smile: 
```php
$isWholeNumber = function($number): bool { return (int) $number == (float) $number; };

if(
  $isWholeNumber($file->importantPartX) && $isWholeNumber($file->importantPartY) &&
  $isWholeNumber($file->importantPartWidth) && $importantPartHeight($file->importantPartX)
) {
  // ...
}
```

Reply:
The `>= 2` check should work for all cases and is simpler IMO so I’d vote for keeping it as it is.

Reply:
Yeah, agree. :smile:  

But for the sake of completeness I'm also throwing [bcadd](https://www.php.net/manual/en/function.bcadd.php) + [bccomp](https://www.php.net/manual/de/function.bccomp.php) in the ring. :monkey: